src/app/components/pages/page.component.ts

```diff
@@ -175,8 +183,68 @@ export abstract class PageComponent extends BaseComponent implements OnInit, OnD
     return openGraphLocaleMap[languageCode];
   }
 
+    /**
+    * Inject the State into the bottom of the <head>
+    */
+   private addLinkTag(tag: LinkDefinition, forceCreation?: boolean) {
+
+     try {
+       const renderer = this.rendererFactory.createRenderer(this.document, {
+         id: '-1',
+         encapsulation: ViewEncapsulation.None,
+         styles: [],
+         data: {}
+       });
+
+       const link = renderer.createElement('link');
+       const head = this.document.head;
+       const selector = this.linkParseSelector(tag);
+
+       if (head === null) {
+         throw new Error('<head> not found within DOCUMENT.');
+       }
+
+       Object.keys(tag).forEach((prop: string) => {
+           return renderer.setAttribute(link, prop, tag[prop]);
+       });
+
+       const children = head.children;
+       for (let i = 0; i < children.length; i++) {
+         if (children[i].localName === 'link' && children[i].rel === tag.rel) {
+           renderer.removeChild(head, children[i]);
+         }
+       }
+       renderer.appendChild(head, link);
+
+     } catch (e) {
+         console.error('Error within linkService : ', e);
+     }
+   }
+
+   private linkParseSelector(tag: LinkDefinition): string {
+       // Possibly re-work this
+       const attr: string = tag.rel ? 'rel' : 'hreflang';
+       return `${attr}="${tag[attr]}"`;
```

Comment:
A bit gnarly code, not obvious what its doing. A comment might be warranted :) (If you dont "re-work" the code as you've already commented 😄)